client-spark/spark2/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -446,6 +446,7 @@ public <K, V> ShuffleWriter<K, V> getWriter(
           this,
           sparkConf,
           shuffleWriteClient,
+          shuffleManagerClient,
```

Comment:
It's your original idea, but I discovered it already initialized in the `RssShuffleManage` construct, so I simply addressed it.

Ok. I will attempt to implement a 'lazyShuffleManagerClient' instead of 'shuffleManagerClient'.

Reply:
+1

Reply:
I have refactored the getReader and getWriter constructs, and I understand why the `RssShuffleManage` should initialize the `shuffleManagerClient` in the constructor. in case of `RssShuffleClientFactory.ExtendWriteClientBuilder` need it. so I also make `ExtendWriteClientBuilder.shuffleManagerClient` is lazy.

Reply:
@advancedxy @zuston PTALA.